aspnetcore/fundamentals/websockets.md

```diff
@@ -37,20 +37,11 @@ This article explains how to get started with WebSockets in ASP.NET Core. [WebSo
 
 * For supported browsers, see https://caniuse.com/#feat=websockets.
 
-## When to use WebSockets
+## NuGet package
```

Comment:
This is removed because SignalR is preferred? If so, why don't we say that in the intro?

Reply:
Replace all MVC action with action to include RP.

Reply:
When to use is removed because it's already covered adequately in the intro, and omitting it lets me promote all the how-to section H3s to H2s.

Reply:
Consider adding 
SignalR is generally preferred over WebSockets
in the intro.

Reply:
That statement would need PU approval.

Reply:
I think that's OK. Might be worth a small amount more detail there though, something like "For most applications, SignalR is generally preferred over raw WebSockets. SignalR provides transport fallback for environments where WebSockets is not available. It also provides a simple "remote procedure call" app model."

I don't have a clean phrasing for it but I'd also make it clear that SignalR should be mostly as efficient as raw WebSockets in almost all environments.